app/assets/javascripts/client/consensus/ConsensusService.js

```diff
@@ -0,0 +1,153 @@
+(function() {
+  'use strict';
+  angular.module('PDRClient')
+      .service('ConsensusService', ConsensusService);
+
+  ConsensusService.$inject = [
+    '$stateParams',
+    '$location',
+    '$q',
+    'Consensus',
+    'AnalysisConsensus',
+    'AnalysisResponse',
+    'Score',
+    'AnalysisConsensusScore',
+    'ProductEntry',
+    'DataEntry'
+  ];
+
+  function ConsensusService($stateParams, $location, $q, Consensus, AnalysisConsensus, AnalysisResponse, Score, AnalysisConsensusScore, ProductEntry, DataEntry) {
+    var service = this;
+
+    service.setContext = function(context) {
+      service.context = context;
+      return service;
+    };
+
+    service.extractId = function() {
+      return $stateParams.assessment_id || $stateParams.analysis_id;
+    };
+
+    service.redirectToIndex = function () {
+      if (service.context === "assessment") {
+        $location.path("/assessments");
+      } else if (service.context === "analysis") {
+        $location.path("/inventories/" + $stateParams.inventory_id + "/analyses");
+      }
+    };
+
+    service.redirectToReport = function () {
+      if (service.context === "assessment") {
+        $location.path("/assessments/" + service.extractId() + "/report");
+      } else if (service.context === "analysis") {
+        $location.path("/inventories/" + $stateParams.inventory_id + "/analyses/" + service.extractId() + "/report");
+      }
+    };
+
+    service.redirectToCreatedConsensus = function (consensusId) {
+      if (service.context === "assessment") {
+        $location.path("/assessments/" + service.extractId() + "/consensus/" + consensusId);
+      } else if (service.context === "analysis") {
+        $location.path("/inventories/" + $stateParams.inventory_id + "/analyses/" + service.extractId() + "/consensus/" + consensusId);
+      }
+    };
+
+    service.createConsensus = function () {
+      if (service.context === "assessment") {
+        return Consensus.create({
+          assessment_id: service.extractId()
+        }, {}).$promise;
+      } else if (service.context === "analysis") {
+        return AnalysisResponse.save({
+          inventory_id: $stateParams.inventory_id,
+          analysis_id: service.extractId()
+        }, {}).$promise
+        .then(function () {
+          return AnalysisConsensus.create({
+            inventory_id: $stateParams.inventory_id,
+            analysis_id: service.extractId()
+          }, {}).$promise;
+        });
+      }
```

Comment:
I consider this sort of "blind" `if`'s a bit dangerous as it will probably increase the effort required to introduce a new context in addition to `assessment` and `analysis`, specially since the method won't have any effect when a new context is used and therefore fail silently since the callers are not really needing the returned promise. Hard to debug.